src/creative.js

```diff
@@ -120,35 +133,96 @@ function renderCrossDomain(adId, pubUrl) {
   requestAdFromPrebid();
 }
 
+/**
+ * Returns cache endpoint concatenated with cache path
+ * @param {string} cacheHost Cache Endpoint host
+ * @param {string} cachePath Cache Endpoint path
+ */
 function getCacheEndpoint(cacheHost, cachePath) {
   let host = (typeof cacheHost === 'undefined' || cacheHost === "") ? DEFAULT_CACHE_HOST : cacheHost;
   let path = (typeof cachePath === 'undefined' || cachePath === "") ? DEFAULT_CACHE_PATH : cachePath;
   return `https://${host}${path}`;
 }
 
-function renderAmpAd(cacheHost, cachePath, uuid) {
-  let adUrl = `${getCacheEndpoint(cacheHost, cachePath)}?uuid=${uuid}`;
+/**
+ * Render mobile or amp ad
+ * @param {string} cacheHost Cache host
+ * @param {string} cachePath Cache path
+ * @param {string} uuid id to render response from cache endpoint
+ * @param {Bool} isMobileApp flag to detect mobile app
+ */
+function renderAmpOrMobileAd(cacheHost, cachePath, uuid, isMobileApp) {
+  // For MoPub, creative is stored in localStorage via SDK.
+  if(uuid.startsWith('Prebid_')) {
+    loadFromLocalCache(uuid)
+  } else {
+    let adUrl = `${getCacheEndpoint(cacheHost, cachePath)}?uuid=${uuid}`;
+    utils.sendRequest(adUrl, handler(isMobileApp));
+  }
+}
 
-  let handler = function(response) {
-    let bidObject;
-    try {
-      bidObject = JSON.parse(response);
-    } catch (error) {
-      console.log(`Error parsing response from cache host: ${error}`);
-    }
-    
+/**
+ * Cache request Callback to display creative
+ * @param {Bool} isMobileApp 
+ */
+function handler(isMobileApp) {
+  return function(response) {
+    let bidObject = parseResponse(response);
     let ad;
-    if (bidObject.adm && bidObject.nurl) {
-      ad = bidObject.adm;
-      ad += utils.createTrackPixelHtml(decodeURIComponent(bidObject.nurl));
-      utils.writeAdHtml(ad);
-    } else if (bidObject.adm) {
-      ad = bidObject.adm;
+    let width = (bidObject.width) ? bidObject.width : bidObject.w;
+    let height = (bidObject.height) ? bidObject.height : bidObject.h;
+    if (bidObject.adm) {
+      ad = (!isMobileApp) ? bidObject.adm : constructMarkup(bidObject.adm, width, height);
```

Comment:
can you swap this logic (don't use !) ? It's confusing to read written this way.